### PR TITLE
Dispose of listeners in quickPickPin and usages

### LINKED ISSUE #225381: Quick pick pin: leaking listeners
Here:

https://github.com/microsoft/vscode/blob/3909bef4e4b26f7b08b187a6332a88ead318678e/src/vs/platform/quickinput/browser/quickPickPin.ts#L24-L32

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
Fixes https://github.com/microsoft/vscode/issues/225381

### CODE REVIEW

**Review by meganrogge — APPROVED** [review]
Thanks!

**Comment by TylerLeonhardt** [comment]
Ill wait for both of you to approve before merging 👍

**Review by songjianet — APPROVED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**
